pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -278,6 +279,11 @@ private synchronized void internalConsumerFlow(Consumer consumer) {
                 log.debug("[{}-{}] Ignoring flow control message since consumer is waiting for cursor to be rewinded",
                         name, consumer);
             }
+        } else if (((ManagedCursorImpl) cursor).resetCursorInProgress()) {
+            if (log.isDebugEnabled()) {
+                log.debug("[{}-{}] Ignoring flow control message since cursor reset in progress - cursor {}",
+                        name, consumer, cursor.getName());
+            }
```

Comment:
This can avoid the new flow permit requests but can't prevent an in-process entry reading operation? The client-side also has an opportunity to receive messages after the seek position.

Reply:
It makes sense. We have to check the seek request status before ack. I have an idea: add a similar status like `resetCursorInProgress `in `ConsumerImpl`. Client has to check the status before sending ack request. How about it ? @codelipenghui